findoutlie/utils.py

```diff
@@ -0,0 +1,96 @@
+
+import nibabel as nib
+import numpy as np
+from matplotlib import pyplot as plt
+
+def basic_stats(image_fname,plot=True):
+    """ Calculate basic stats metrics to be reused in different metrics
+    for outlier detection
+
+    Parameters
+    ----------
+    image_fname : path to a nii image, like r".\data\group-00\sub-01\func\sub-01_task-taskzero_run-01_bold.nii.gz"
+    plot : boolean, true or false, true if you want the nii as outputs to check the function
+
+    Returns
+    -------
+    mean_data : 3D array, nii image (x,y,z axis)
+        Mean of the image over time
+    sd_data : 3D array, nii image (x,y,z axis)
+        Temporal deviation map, Standard deviation of the image over time
+
+    meanmap_median : float, median of the mean nifti
+    sdmap_median : float, median of the sd nifti
+    """
+
+    filename,img,data=get_image(image_fname)
+
+    mean_data = np.mean(data,axis=-1)
+    #mean of image on the 4th dim, time
+    sd_data=np.std(data,axis=-1) #temporal deviation map
+
+    # Median #make sense ?
+    meanmap_median=np.median(mean_data)
+    sdmap_median = np.median(sd_data)
+
+    if plot:
```

Comment:
I don't think I really understand why we would save the metrics data in a Nifti1 image. Shouldn't the metrics be a 1D vector of lenght `n_timepoints` or less?

Reply:
Here it was not a final metric, just an intermediate computation for other metrics. The value was across time I think, voxel-wise. I agree it will not help so much to find outlier to have the nii but it helped me check & visualize what I was doing (as a beginner in fmri coding ;) ). This is why "plot" was an argument but it should always be set up to "False" I guess indeed. Thanks Alex !